### PR TITLE
[ADVISOR-2860] Add status column to Activity table

### PR DESCRIPTION
**Pull request by johnsonm325**
This PR adds a status column to the Activity table with an icon for Completed and Running tasks.

### CODE REVIEW

File: src/SmartComponents/ActivityTable/Columns.js [hunk 1]
```diff
@@ -15,10 +16,34 @@ TaskNameCell.propTypes = {
   index: propTypes.number,
 };
 
+const StatusCell = ({ status }) => {
+  if (status === 'Completed') {
```
Comment (mkholjuraev) [inline src/SmartComponents/ActivityTable/Columns.js:20]: How about using a switch statement?

File: src/SmartComponents/ActivityTable/__tests__/ActivityTable.tests.js [hunk 2]
```diff
@@ -102,9 +118,61 @@ describe('CompletedTasksTable', () => {
...
+
+    render(
+      <MemoryRouter keyLength={0}>
+        <Provider store={store}>
+          <ActivityTable />
+        </Provider>
+      </MemoryRouter>
+    );
+
+    await waitFor(() => expect(fetchExecutedTasks).toHaveBeenCalled());
```
Comment (mkholjuraev) [inline src/SmartComponents/ActivityTable/__tests__/ActivityTable.tests.js:134]: I believe we can use a single waitFor for all the `expect`s. In other words, we can group all the expected asynchronous calls inside a single waitFor.

Comment (johnsonm325) [reply src/SmartComponents/ActivityTable/__tests__/ActivityTable.tests.js]: This works for most cases, but there is a test in this file that fails, so I left the test as-is.

File: src/SmartComponents/ActivityTable/Columns.js (hunk 1 above)
Comment (bastilian) [reply src/SmartComponents/ActivityTable/Columns.js:20]: You could also go fancy pants and avoid if/else and switch:

```js
const StatusCell = ({ status }) => ({
    Completed: (
        <span style={{ color: '#3E8635', display: 'flex', alignItems: 'center' }}>
            <CheckCircleIcon color="#3E8635" style={{ marginRight: '4px' }} />
            {status}
        </span>
    ),
    Running: (
        <span style={{ color: '#2B9AF3', display: 'flex', alignItems: 'center' }}>
            <InProgressIcon color="#2B9AF3" style={{ marginRight: '4px' }} />
            {status}
        </span>
    )
}[status] || status);
```

However, i think the if/else here is fine, would there be more a mapped object like ^ would be the neatest solution.

File: src/SmartComponents/ActivityTable/ActivityTable.js [hunk 3]
```diff
@@ -56,13 +53,11 @@ const CompletedTasksTable = () => {
         setTaskError
       );
 
-      if (fetchedTaskJobs.length) {
-        setSelectedSystems(getSelectedSystems(fetchedTaskJobs));
-        await setCompletedTaskDetails(fetchedTaskDetails);
-      }
+      setSelectedSystems(getSelectedSystems(fetchedTaskJobs));
+      await setActivityDetails(fetchedTaskDetails);
```
Comment (mkholjuraev) [inline src/SmartComponents/ActivityTable/ActivityTable.js:57]: I think `await` can be safely removed as fetchedTaskDetails in not a promise.

File: src/SmartComponents/ActivityTable/ActivityTable.js [hunk 4]
```diff
@@ -56,13 +53,11 @@ const CompletedTasksTable = () => {
...
     } else {
       setRunTaskModalOpened(false);
-      await setCompletedTaskDetails({});
+      await setActivityDetails({});
```
Comment (mkholjuraev) [inline src/SmartComponents/ActivityTable/ActivityTable.js:60]: Here also

File: src/SmartComponents/ActivityTable/Columns.js [hunk 5]
```diff
@@ -15,10 +16,30 @@ TaskNameCell.propTypes = {
   index: propTypes.number,
 };
 
+const StatusCell = ({ status }) =>
```
Comment (mkholjuraev) [inline src/SmartComponents/ActivityTable/Columns.js:19]: Looks great now!

**Review by mkholjuraev — APPROVED** [review]
LGTM! The status column is present in the Activities tab, sortable. The table's basic functionality is working as expected. Just a minor thing that needs to be removed.

**Pull request merged**